### PR TITLE
[0D-Tensor] CINN supports `fill_constant`, fix infershape and pass

### PR DESCRIPTION
**Pull request by jiahy0825**
### PR types
Function optimization

### PR changes
OPs

### Description

Pcard-71127

[0D-Tensor] CINN supports `fill_constant`, fix infershape and pass

- Link to the first PR: #53382

### CODE REVIEW

File: paddle/cinn/frontend/pass/expand_zero_dim_pass.cc [hunk 1]
```diff
@@ -101,6 +104,20 @@ class ExpandZeroDimPass : public ProgramPass {
     }
     return new_instr;
   }
+
+  // Before: out-0D = fill_constant([], 123.456, "out", "float32")
+  // After:  out-1D = fill_constant([1], 123.456, "out", "float32")
+  Instruction HandleFillConstant(const Instruction& instr) {
+    Instruction new_instr = instr;
```
Comment (zyfncg) [inline paddle/cinn/frontend/pass/expand_zero_dim_pass.cc:111]: 这里为什么转成了1D？

Comment (jiahy0825) [reply paddle/cinn/frontend/pass/expand_zero_dim_pass.cc:111]: 这个 pass 会把对 0D-Tensor 的处理转为 1D-Tensor，从而适配后端生成逻辑

Comment (zyfncg) [reply paddle/cinn/frontend/pass/expand_zero_dim_pass.cc:111]: 那就是后端的代码生成还不支持`fill_constant`的0D-Tensor？

Comment (jiahy0825) [reply paddle/cinn/frontend/pass/expand_zero_dim_pass.cc:111]: 是的

**Review by zyfncg — APPROVED** [review]

**Pull request merged**